### ISSUE TITLE
Source maps with inlined source

### ISSUE THREAD

**Issue by weinand** [Microsoft/vscode#2183]
- [x] win @SofianHn 
- [x] mac @joaomoreno 
- [x] linux @aeschli 

Node debugging now supports source maps with "inlined source" (Please find the definition of "inlined source" vs. "inlined source maps" below).

Here is the strategy in what situations VSCode will use "inlined source":
VSCode always tries to locate the source on disk first. If it cannot find the source (e.g. because there is none or because the paths in the source maps are broken), VSCode will use the "inlined source" if available. If there is no inlined source, VSCode will fall back to stream the file contents from node itself.

If the editor contents is not loaded by VSCode from the file system but comes from the debug adapter, the editor will be in readonly mode and the "origin" of the editor contents is shown in the editor title:

![2016-01-22 16-24-31](https://cloud.githubusercontent.com/assets/1898161/12514672/cadac9a0-c124-11e5-9838-b09bdf57c45a.png)

**Test:** For your favourite project create source maps with inlined source.
- verify that VSCode uses the strategy from above for showing the source
- verify that you can step through inlined source
- verify that breakpoints can be set in inlined source and that they are persisted
- combine inlined source and inlined source maps and verify that still everything works.
 

**Note:** "inlined source" and "inlined source maps" are orthogonal features (and VSCode supports both either alone or in combination):
- *inlined source maps:* the contents of the source map does not live in a file but is a data url at the end of the generated file
- *inlined source:* the contents of the original source file does not live in a file but is included in the source map.

**Comment by joaomoreno** [comment]
This took some quite work to test... You need to debug an application that has both a file in your workspace (so you are able to set a breakpoint and thus pause the debugger) and a sourcemap'ed file which needs to appear in the call stack during that pause, otherwise you can never see it. 

In order to save time for the others, here's some sample code: https://gist.github.com/joaomoreno/a1f68c4b430c8bd40d61

Simply put that in your disk, open the folder with VSCode, start debugging `main.js` with sourcemaps enabled. You can put a breakpoint on line 4 step in once its hit.

**Comment by weinand** [comment]
@joaomoreno thanks a lot! BTW, another way to test this is by enabling "stop on entry".

**Comment by joaomoreno** [comment]
Ahhh forgot about that!

**Comment by aeschli** [comment]
I verified on Linux with Joao's example
